genesyscloud/supported_content/resource_genesyscloud_supported_content_utils.go

```diff
@@ -0,0 +1,115 @@
+package supported_content
+
+import (
+	"terraform-provider-genesyscloud/genesyscloud/util/resourcedata"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/mypurecloud/platform-client-sdk-go/v133/platformclientv2"
+)
+
+/*
+The resource_genesyscloud_supported_content_utils.go file contains various helper methods to marshal
+and unmarshal data into formats consumable by Terraform and/or Genesys Cloud.
+*/
+
+// getSupportedContentFromResourceData maps data from schema ResourceData object to a platformclientv2.Supportedcontent
+func getSupportedContentFromResourceData(d *schema.ResourceData) platformclientv2.Supportedcontent {
+	return platformclientv2.Supportedcontent{
+		Name:       platformclientv2.String(d.Get("name").(string)),
+		MediaTypes: buildMediaTypes(d.Get("media_types").([]interface{})),
+	}
+}
```

Comment:
The way this function is working, if the user only defines a value for `name`, the MediaTypes field will be defined as well but with nothing inside, resulting in the error: `Failed to create supported content: API Error: 400 - Missing the supportedContent.mediaTypes.allow field`

We want to make sure MediaTypes is nil if the user doesn't specify it in their resource. You can do this by defining `platformclientv2.Supportedcontent` as a variable 

```golang
var supportedContent platformclientv2.Supportedcontent
supportedContent.name = platformclientv2.String(d.Get("name").(string))
if mediaTypes, ok := d.Get("media_types").([]interface{}); ok && len(mediaTypes) > 0 {
  supportedContent.MediaTypes = &mediaTypes 
}
return supportedContent
```